clippy_lints/src/error_impl_error.rs

```diff
@@ -0,0 +1,76 @@
+use clippy_utils::{
+    diagnostics::{span_lint, span_lint_hir_and_then},
+    path_res,
+    ty::implements_trait,
+};
+use rustc_hir::{def_id::DefId, Item, ItemKind, Node};
+use rustc_hir_analysis::hir_ty_to_ty;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::sym;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for types named `Error` that implement `Error`.
+    ///
+    /// ### Why is this bad?
+    /// It can become confusing when a codebase has 20 types all named `Error`, requiring either
+    /// aliasing them in the `use` statement them or qualifying them like `my_module::Error`. This
+    /// severely hinders readability.
```

Comment:
This isn't really a much of a readability. The alternative name would be `MyModuleError` or `MyCrateError`, which isn't really any worse to read.

Reply:
I'd say it's more comprehension, which is what I was *trying* to get at. Having a consistent name for a type is better than sometimes using `my_module::Error`, `MyModuleError` or just `Error`. Similar, `missing_enforced_import_renames`